### PR TITLE
fix: make sure 3.12 is the system/default python version

### PR DESCRIPTION
**Pull request by kaltepeter**
I finally figured out why I was getting 3.11 as the default python version. `pipx` used the default debian version instead of the system/base container version. I moved pipx install out of apt-get and it resolved.

```bash
python@2f8648140955:/usr/src$ pipx environment
Environment variables (set by user):

PIPX_HOME=
PIPX_GLOBAL_HOME=
PIPX_BIN_DIR=
PIPX_GLOBAL_BIN_DIR=
PIPX_MAN_DIR=
PIPX_GLOBAL_MAN_DIR=
PIPX_SHARED_LIBS=
PIPX_DEFAULT_PYTHON=
PIPX_FETCH_MISSING_PYTHON=
USE_EMOJI=
PIPX_HOME_ALLOW_SPACE=

Derived values (computed by pipx):

PIPX_HOME=/home/python/.local/share/pipx
PIPX_BIN_DIR=/home/python/.local/bin
PIPX_MAN_DIR=/home/python/.local/share/man
PIPX_SHARED_LIBS=/home/python/.local/share/pipx/shared
PIPX_LOCAL_VENVS=/home/python/.local/share/pipx/venvs
PIPX_LOG_DIR=/home/python/.local/state/pipx/log
PIPX_TRASH_DIR=/home/python/.local/share/pipx/trash
PIPX_VENV_CACHEDIR=/home/python/.cache/pipx
PIPX_STANDALONE_PYTHON_CACHEDIR=/home/python/.local/share/pipx/py
PIPX_DEFAULT_PYTHON=/usr/local/bin/python3.12
USE_EMOJI=true
PIPX_HOME_ALLOW_SPACE=false
python@2f8648140955:/usr/src$ ls -la /home/python/.local/share/pipx/venvs
total 16
drwxr-xr-x 4 python python 4096 Dec 20 19:41 .
drwxr-xr-x 5 python python 4096 Dec 20 19:41 ..
drwxr-xr-x 5 python python 4096 Dec 20 19:41 nox
drwxr-xr-x 5 python python 4096 Dec 20 19:41 poetry
python@2f8648140955:/usr/src$ ls -la /home/python/.local/share/pipx/venvs/poetry/
total 28
drwxr-xr-x 5 python python 4096 Dec 20 19:41 .
drwxr-xr-x 4 python python 4096 Dec 20 19:41 ..
drwxr-xr-x 2 python python 4096 Dec 20 19:41 bin
drwxr-xr-x 3 python python 4096 Dec 20 19:41 include
drwxr-xr-x 3 python python 4096 Dec 20 19:41 lib
lrwxrwxrwx 1 python python    3 Dec 20 19:38 lib64 -> lib
-rw-r--r-- 1 python python 3035 Dec 20 19:39 pipx_metadata.json
-rw-r--r-- 1 python python  217 Dec 20 19:38 pyvenv.cfg
python@2f8648140955:/usr/src$ cat /home/python/.local/share/pipx/venvs/poetry/pyvenv.cfg 
home = /usr/local/bin
include-system-site-packages = false
version = 3.12.8
executable = /usr/local/bin/python3.12
command = /usr/local/bin/python3.12 -m venv --without-pip /home/python/.local/share/pipx/venvs/poetry
python@2f8648140955:/usr/src$ exit
exit
```

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ LOCAL_NAME=${DOCKER_REPO}/datasite/test-cicd-python
 
 echo "Building ${LOCAL_NAME}"
 
-docker buildx build --builder=container --platform="${PLATFORMS}" --push --pull -t "${LOCAL_NAME}:latest-python" "${__dir}/"
+docker buildx build --progress plain --no-cache --builder=container --platform="${PLATFORMS}" --push --pull -t "${LOCAL_NAME}:latest-python" "${__dir}/"
```
Comment (kaltepeter) [inline build.sh:14]: caching during build can cause trouble

File: generate-test-nx.sh [hunk 2]
```diff
@@ -6,9 +6,11 @@ set -o nounset
 readonly __dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
 OUTPUT_DIR=test-nx
+rm -rf "${OUTPUT_DIR}"
 
 npx create-nx-workspace ${OUTPUT_DIR} --preset=npm --nxCloud=skip
 cp ./Dockerfile.test-nx ./${OUTPUT_DIR}/Dockerfile
+cp ./test-app/poetry.toml ./test-nx/poetry.toml
```
Comment (kaltepeter) [inline generate-test-nx.sh:13]: make sure poetry in in workspace to avoid a global definition of env

File: test-nx/README.md [hunk 3]
```diff
@@ -2,75 +2,107 @@
 
 <a alt="Nx logo" href="https://nx.dev" target="_blank" rel="noreferrer"><img src="https://raw.githubusercontent.com/nrwl/nx/master/images/nx-logo.png" width="45"></a>
 
-✨ **This workspace has been generated by [Nx, Smart Monorepos · Fast CI.](https://nx.dev)** ✨
+✨ Your new, shiny [Nx workspace](https://nx.dev) is ready ✨.
```
Comment (kaltepeter) [inline test-nx/README.md:5]: all generated

File: test-nx/package.json [hunk 4]
```diff
@@ -1,15 +1,16 @@
 {
-  "name": "test-nx",
+  "name": "@test-nx/source",
```
Comment (kaltepeter) [inline test-nx/package.json:2]: generated

File: test-nx/packages/proj1/project.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
   "name": "proj1",
   "$schema": "../../node_modules/nx/schemas/project-schema.json",
   "projectType": "application",
-  "sourceRoot": "packages/proj1/pymonorepo_proj1",
+  "sourceRoot": "packages/proj1/proj1",
```
Comment (kaltepeter) [inline test-nx/packages/proj1/project.json:5]: i don't care for the double name, but it tests the thing that needs testing

**Review by NickBeeuwsaert-Datasite — APPROVED** [review]

**Pull request merged**